recipes-core/tdx-enc-handler/tdx-enc-handler/tdx-enc.sh

```diff
@@ -97,6 +124,63 @@ tdx_enc_prepare_tpm() {
     fi
 }
 
+tdx_enc_key_recover_from_partition() {
+    tdx_enc_log "Recovering encrypted key blob from partition ${TDX_ENC_STORAGE_LOCATION}..."
+
+    TDX_ENC_KEY_FULLPATH="/tmp/${TDX_ENC_KEY_FILE}"
+    rm -rf $TDX_ENC_KEY_FULLPATH
+
+    STORAGE_KEY_BLOCK_DATA=/tmp/.enckey.txt
+    if ! dd if=${TDX_ENC_STORAGE_LOCATION} of=${STORAGE_KEY_BLOCK_DATA} skip=${TDX_ENC_STORAGE_NUM_BLOCKS} bs=512 count=1; then
+        tdx_enc_exit_error "Could not read block from ${TDX_ENC_STORAGE_LOCATION} with key information!"
+    fi
+
+    while read -r line; do
+        key=$(echo "$line" | cut -d'=' -f1)
+        val=$(echo "$line" | cut -d'=' -f2)
+        case "$key" in
+            "keyname") keyname="${val}" ;;
+            "keydata") keydata="${val}" ;;
+            "keycsum") keycsum="${val}" ;;
+        esac
+    done < ${STORAGE_KEY_BLOCK_DATA}
```

Comment:
Hey @sergioprado,

I did a couple tests and found that this loop will happily process the data after the EOT:

Running this:

```
{
    printf "key1=val1\n"
    printf "key2=\tstring with tabs and NUL \0\0\0\0 characters !\n"
    printf "key3=val3\n"
    printf "\04"
    printf "badkey1=data after EOT"
    printf "\0\0\0\0\0\0\0\0\0\0\0\0\n"
    printf "badkey2=data after EOT\n"
} > tt.tmp

while read -r line; do
    echo "line: $line"
done < tt.tmp
```

I got the output:

```
line: key1=val1
line: key2=	string with tabs and NUL  characters !
line: key3=val3
line: ^Dbadkey1=data after EOT
line: badkey2=data after EOT
```

A simple solution is (tested on bash, dash and busybox sh):

```
EOT=$(printf '\004')
while read -r line; do
    echo "$line" | grep -q "$EOT" && break
    echo "line: $line"
done < tt.tmp
```

where the output is as expected:

```
line: key1=val1
line: key2=	string with tabs and NUL  characters !
line: key3=val3
```